src/lib/components/primitives/input/floating-label.svelte

```diff
@@ -1,24 +1,34 @@
 <script lang="ts">
+  export let hasRightIcon = false;
   export let key: string;
   export let label: string;
   export let onChange: (event: Event) => void;
+  export let isRequired = false;
+  // TODO: Placeholders don't reliably work with floating labels
+  // export let placeholder: string;
   export let type: 'date' | 'email' | 'number' | 'password' | 'phone' | 'text' = 'text';
   export let value: string | null = null;
   export let width: 'full' | 'half' = 'full';
 </script>
 
-<div class="tw_relative">
+<div class={`width-${width} tw_items-stretch tw_flex tw_mb-4 tw_relative tw_flex-wrap`}>
+  <!--
+    TODO: Improve layout of floating label. Maybe use grid instead of Twitter's CSS?
+    NOTE: `placeholder` for the input is required for Twitter Bootstrap's floating label
+  -->
   <input
+    class={`width-{width} tw_border tw_border-gray ${hasRightIcon ? 'tw_border-r-0': ''} tw_bg-white tw_block  tw_flex-1 tw_basis-auto tw_leading-6 tw_p-3 tw_rounded ${hasRightIcon ? 'tw_rounded-r-none' : ''} tw_text-base tw_text-gray-dark tw_w-1`}
     id={key}
-    {type}
     on:change={onChange}
-    class="width-{width} tw_border tw_border-gray tw_bg-white tw_block tw_leading-6 tw_mb-4 tw_p-3 tw_rounded tw_text-base tw_text-gray-dark tw_w-full"
-    {value}
     placeholder={label}
+    {type}
```

Comment:
I don't see any validation in the storybook when type is set to email, is this intended?

Reply:
Good question. I believe validation from a native input perspective happens at form submission, not on blur or on change.